src/app/modules/users/users.service.ts

```diff
@@ -1,21 +1,60 @@
-import { Injectable } from '@nestjs/common';
-import { InjectRepository } from '@nestjs/typeorm';
+import {HttpException, HttpStatus, Injectable} from '@nestjs/common';
 import { v4 } from 'uuid';
 import { UserEntity } from '@/app/modules/users/user.entity';
-import { UsersRepository } from '@/app/modules/users/users.repository';
 import { OauthProvider } from '@/app/modules/common/enums/provider.enum';
+import UsersListResponseDto from '@/app/modules/users/dto/user-item.response.dto';
+import {UsersRepository} from '@/app/modules/users/users.repository';
+import {UserStatusEnum} from '@/app/modules/common/enums/user-status.enum';
+import {EntityManager} from 'typeorm';
+import {PaginateResponseDto} from '@/app/response/dto/paginate-response.dto';
+import {UserPaginatorDto} from '@/app/modules/users/dto/user-paginator.dto';
+import {Language} from '@/app/enum/language.enum';
 
 @Injectable ()
 export class UsersService {
   constructor (
-    @InjectRepository (UserEntity)
-    private readonly usersRepository: UsersRepository,
+      private readonly usersRepository: UsersRepository,
+      private readonly entityManager: EntityManager
   ) {
   }
 
```

Comment:
Renive async/await for all this methods, since you just return promise here.
https://eslint.org/docs/latest/rules/no-return-await